src/module-elasticsuite-virtual-category/Model/Rule.php

```diff
@@ -139,15 +139,21 @@ public function __toString(): string
     public function getCategorySearchQuery($category, $excludedCategories = []): ?QueryInterface
     {
         $query         = null;
+        $rootCategory  = null;
```

Comment:
I don't a reason to initialize that variable since it is systematically overwritten in your additional condition.